base/compiler/abstractinterpretation.jl

```diff
@@ -887,6 +887,12 @@ function abstract_call_method_with_const_args(interp::AbstractInterpreter,
     if !const_prop_enabled(interp, sv, match)
         return nothing
     end
+    if is_removable_if_unused(result.effects)
```

Comment:
Can we also bail out if the return type is Bottom?

Reply:
I think if the `rt` is known to be Union{}, we need to skip concrete eval and let the regular heuristics take care of it, which disable const prop for this case unless explicitly forced (which could make sense to simplify the inlined IR).

Reply:
That's mostly unrelated to this PR though, so we should probably merge this first and then I can put up a separate PR for that change.